plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestTrinoRestCatalog.java

```diff
@@ -200,4 +200,31 @@ public void testPrefix()
                 .as("should fail as the prefix dev is not implemented for the current endpoint")
                 .hasMessageContaining("Malformed request: No route for request: POST v1/dev/namespaces");
     }
+
+    @Test
+    public void testMultipartNamespace()
+    {
+        String dotSeparator = ".";
+        TrinoCatalog catalog = createTrinoCatalog(false);
+
+        String firstLevel = "level-1" + randomNameSuffix();
+        String secondLevel = "level-2" + randomNameSuffix();
+        String thirdLevel = "level-3" + randomNameSuffix();
+
+        ImmutableList<String> namespacesList = ImmutableList.of(
+                firstLevel,
+                firstLevel + dotSeparator + secondLevel,
+                firstLevel + dotSeparator + secondLevel + dotSeparator + thirdLevel);
+
+        namespacesList.forEach(namespace -> catalog.createNamespace(SESSION, namespace, ImmutableMap.of(), new TrinoPrincipal(PrincipalType.USER, SESSION.getUser())));
+
+        try {
+            assertThat(catalog.listNamespaces(SESSION)).as("catalog.listNamespaces")
```

Comment:
Verifying only `listNamespaces` is insufficient. We should check all statements with query-based tests instead of calling this method. For instance, the following statement succeeds on master, but fails on this PR:
```
trino>  CREATE SCHEMA iceberg."a.test";
Query 20240926_230804_00025_uiqvm failed: Malformed request: <html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1"/>
<title>Error 400 Suspicious Path Character</title>
</head>
<body>
<h2>HTTP ERROR 400 Suspicious Path Character</h2>
<table>
<tr><th>URI:</th><td>/badURI</td></tr>
<tr><th>STATUS:</th><td>400</td></tr>
<tr><th>MESSAGE:</th><td>Suspicious Path Character</td></tr>
</table>
<h3>Caused by:</h3><pre>org.eclipse.jetty.http.BadMessageException: 400: Suspicious Path Character
	at org.eclipse.jetty.server.internal.HttpConnection$HttpStreamOverHTTP1.headerComplete(HttpConnection.java:1199)
	at org.eclipse.jetty.server.internal.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:964)
	at org.eclipse.jetty.http.HttpParser.parseFields(HttpParser.java:1345)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1641)
	at org.eclipse.jetty.server.internal.HttpConnection.parseRequestBuffer(HttpConnection.java:563)
	at org.eclipse.jetty.server.internal.HttpConnection.onFillable(HttpConnection.java:386)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:322)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:99)
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:478)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:441)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:293)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:201)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:311)
	at org.eclipse.jetty.util.thread.MonitoredQueuedThreadPool$1.run(MonitoredQueuedThreadPool.java:73)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:979)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1209)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1164)
	at java.base/java.lang.Thread.run(Thread.java:1570)
</pre>

</body>
</html>
```

Reply:
@ebyhr, when I run this query, I get different result:
```
trino> create schema "a.test";
Query 20240927_194218_00013_fr4es failed: Namespace does not exist: a
```
Here's the exception from the query overview:
```
org.apache.iceberg.exceptions.NoSuchNamespaceException: Namespace does not exist: a
	at org.apache.iceberg.rest.ErrorHandlers$NamespaceErrorHandler.accept(ErrorHandlers.java:173)
	at org.apache.iceberg.rest.ErrorHandlers$NamespaceErrorHandler.accept(ErrorHandlers.java:166)
	at org.apache.iceberg.rest.HTTPClient.throwFailure(HTTPClient.java:211)
	at org.apache.iceberg.rest.HTTPClient.execute(HTTPClient.java:323)
	at org.apache.iceberg.rest.HTTPClient.execute(HTTPClient.java:262)
	at org.apache.iceberg.rest.HTTPClient.post(HTTPClient.java:368)
	at org.apache.iceberg.rest.RESTClient.post(RESTClient.java:112)
	at org.apache.iceberg.rest.RESTSessionCatalog.createNamespace(RESTSessionCatalog.java:538)
	at io.trino.plugin.iceberg.catalog.rest.TrinoRestCatalog.createNamespace(TrinoRestCatalog.java:198)
	at io.trino.plugin.iceberg.IcebergMetadata.createSchema(IcebergMetadata.java:876)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.createSchema(ClassLoaderSafeConnectorMetadata.java:410)
	at io.trino.tracing.TracingConnectorMetadata.createSchema(TracingConnectorMetadata.java:340)
	at io.trino.metadata.MetadataManager.createSchema(MetadataManager.java:787)
	at io.trino.tracing.TracingMetadata.createSchema(TracingMetadata.java:374)
	at io.trino.execution.CreateSchemaTask.internalExecute(CreateSchemaTask.java:128)
	at io.trino.execution.CreateSchemaTask.execute(CreateSchemaTask.java:82)
	at io.trino.execution.CreateSchemaTask.execute(CreateSchemaTask.java:54)
	at io.trino.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:146)
	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:272)
	at io.trino.dispatcher.LocalDispatchQuery.startExecution(LocalDispatchQuery.java:150)
	at io.trino.dispatcher.LocalDispatchQuery.lambda$waitForMinimumWorkers$2(LocalDispatchQuery.java:134)
	at io.airlift.concurrent.MoreFutures.lambda$addSuccessCallback$12(MoreFutures.java:570)
	at io.airlift.concurrent.MoreFutures$3.onSuccess(MoreFutures.java:545)
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1137)
	at io.trino.$gen.Trino_a6e235a____20240927_192839_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
```
@mayankvadariya and I decided to add more tests to the respective catalogs to cover different objects creation/deletion under the nested namespaces. Do you think this would be still insufficient?

Reply:
I've added query based tests

Reply:
@sftwrdvlpr You can reproduce this issue with `IcebergRestQueryRunnerMain`.

Reply:
@ebyhr I found a solution to this issue by calling a different `TestingHttpServer` constructor and setting `enableLegacyUriCompliance` to `true`. If this approach looks good to you, I'll proceed with adding the change to the PR.

Reply:
@ebyhr I've added changes, seems all good